### ISSUE TITLE
Various reward-related flaky tests

### ISSUE THREAD

**Issue by Anviking** [cardano-foundation/cardano-wallet#2467]
# Context
 
 

#2449 recently changed core logic in the network layer.

Very rarely, this failure appeared on the branch:
```
  src/Test/Integration/Framework/DSL.hs:1796:7:
  2) API Specifications, SHELLEY_STAKE_POOLS, STAKE_POOLS_JOIN_01rewards - Can join a pool, earn rewards and collect them
       expected a successful response but got an error: DecodeFailure "{\"code\":\"created_invalid_transaction\",\"message\":\"That's embarrassing. It looks like I've created an invalid transaction that could not be parsed by the node. Here's an error message that may help with debugging: HardForkApplyTxErrFromEra S (S (S (Z (WrapApplyTxErr {unwrapApplyTxErr = ApplyTxError [LedgerFailure (DelegsFailure (WithdrawalsNotInRewardsDELEGS (fromList [(RewardAcnt {getRwdNetwork = Mainnet, getRwdCred = KeyHashObj (KeyHash \\\"56e36e911d28670c50eaf7169e507bc1422162d3eb8a3c9982e2f95d\\\")},Coin 19675454185)])))]}))))\"}"
       While verifying (Status {statusCode = 500, statusMessage = "Internal Server Error"},Left (DecodeFailure "{\"code\":\"created_invalid_transaction\",\"message\":\"That's embarrassing. It looks like I've created an invalid transaction that could not be parsed by the node. Here's an error message that may help with debugging: HardForkApplyTxErrFromEra S (S (S (Z (WrapApplyTxErr {unwrapApplyTxErr = ApplyTxError [LedgerFailure (DelegsFailure (WithdrawalsNotInRewardsDELEGS (fromList [(RewardAcnt {getRwdNetwork = Mainnet, getRwdCred = KeyHashObj (KeyHash \\\"56e36e911d28670c50eaf7169e507bc1422162d3eb8a3c9982e2f95d\\\")},Coin 19675454185)])))]}))))\"}"))

  To rerun use: --match "/API Specifications/SHELLEY_STAKE_POOLS/STAKE_POOLS_JOIN_01rewards - Can join a pool, earn rewards and collect them/"
```
though I have also seen it _once_ before. 

After merging it, there seem to be an uptick in failures related to withdrawals.

Need to investigate, but _might_ be that the reward balance is less consistently updated when the wallet is under high load (as in integration tests), since #2449.

# Test Case

Multiple, related to withdrawals.

# Failure / Counter-example

---

# Resolution 

---

# QA

**Comment by Anviking** [comment]
<img width="1052" alt="Skärmavbild 2021-06-14 kl  21 30 43" src="https://user-images.githubusercontent.com/304423/121948671-d12a6f00-cd57-11eb-8e20-12fcf132b2b6.png">

Hoping this is fixed by #2698. Can re-open if not.

**Comment by jonathanknowles** [comment]
Saw a similar failure on this build:
https://buildkite.com/cardano-foundation/cardano-wallet/builds/827#018a695b-5e98-4c2c-b79c-824cb7fac8a6/6-708
